### PR TITLE
Restore remap boxes

### PR DESCRIPTION
**Pull request by Rob192**
This PR is to address #800 as part of #788. 

It restores the function `remap_boxes` from #488 and adds a new functionality test to verify the output of a pretrained predictor on a rotated document.

### CODE REVIEW

File: doctr/utils/geometry.py [hunk 1]
```diff
@@ -127,11 +127,41 @@ def rotate_abs_geoms(
     return rotated_polys
 
 
+def remap_boxes(
+    loc_preds: np.ndarray,
+    orig_shape: Tuple[int, int],
+    dest_shape: Tuple[int, int]
+) -> np.ndarray:
+    """ Remaps a batch of rotated locpred (x, y, w, h, alpha, c) expressed for an origin_shape to a destination_shape.
```
Comment (charlesmindee) [inline doctr/utils/geometry.py:135]: Is this a typo in the docstring ? The function should work with the new box format [[x1, y1], [x2, y2], ...]]

File: doctr/utils/geometry.py [hunk 2]
```diff
@@ -127,11 +127,41 @@ def rotate_abs_geoms(
...
+    This does not impact the absolute shape of the boxes, but allow to calculate the new relative RotatedBbox
+    coordinates after a resizing of the image.
+    Args:
+        loc_preds: (N, 6) array of RELATIVE locpred (x, y, w, h, alpha, c)
+        orig_shape: shape of the origin image
+        dest_shape: shape of the destination image
+    Returns:
+        A batch of rotated loc_preds (N, 6): (x, y, w, h, alpha, c) expressed in the destination referencial
```
Comment (charlesmindee) [inline doctr/utils/geometry.py:143]: same here, you need to return boxes in the new format

File: doctr/utils/geometry.py [hunk 3]
```diff
@@ -127,11 +127,41 @@ def rotate_abs_geoms(
...
+    """
+
+    if len(dest_shape) != 2:
+        raise ValueError(f"Mask length should be 2, was found at: {len(dest_shape)}")
+    if len(orig_shape) != 2:
+        raise ValueError(f"Image_shape length should be 2, was found at: {len(orig_shape)}")
+    orig_height, orig_width = orig_shape
+    dest_height, dest_width = dest_shape
+    mboxes = loc_preds.copy()
+    mboxes[:, :, 0] = ((loc_preds[:, :, 0] * orig_width) + (dest_width - orig_width) / 2) / dest_width
```
Comment (charlesmindee) [inline doctr/utils/geometry.py:153]: I am not sure I understand well the computation here: If we have relative coordinates [[x1, y1], [x2, y2], [x3, y3], ...] we just need to multiply by dest_width & dest_height to get the absolute points coordinates in dest_shape referential, or am I missing something ?

**Review by charlesmindee — COMMENTED** [review]
Hi @Rob192, thank you for your PR. 
I added a few comments because I am not sure I fully understood the purpose of the remap_boxes function since we moved to a new format of boxes where boxes are described by points and not angles. In any case, can you attach in the issue #800 the image of the document where the boxes are misplaced to help us understanding the bug ? :pray: Thank you!

File: doctr/utils/geometry.py (hunk 3 above)
Comment (Rob192) [reply doctr/utils/geometry.py:153]: Hi @charlesmindee, no you need to take into account the padding that was introduced by `rotate_image`. This is the main point of this function `remap_boxes` that is here to accommodate these differences.

File: doctr/utils/geometry.py (hunk 2 above)
Comment (Rob192) [reply doctr/utils/geometry.py:143]: indeed, this is a typo !

File: doctr/utils/geometry.py (hunk 1 above)
Comment (Rob192) [reply doctr/utils/geometry.py:135]: typo indeed

File: doctr/utils/geometry.py (hunk 3 above)
Comment (Rob192) [reply doctr/utils/geometry.py:153]: I added pictures for reference in #800

Comment (charlesmindee) [reply doctr/utils/geometry.py:153]: Ok, thanks I see the point

**Review by charlesmindee — APPROVED** [review]
thanks for this fix!

**Pull request merged**